src/client/components/layouts/Header/NavItemWithSubItems.tsx

```diff
@@ -76,64 +82,71 @@ export function NavItemWithSubItems({ className, onClick, item: root }: NavItemW
 
-	useEffect(() => {
-		for (const item of root.children) {
-			if (!isNavigationItem(item) && isItemActive(item, router.pathname)) {
-				selectEmbeddedNavItem(item);
-				break;
-			}
+	const subNav = currentItem.children.map((item, index) => {
+		if (isNavigationItem(item)) {
+			return (
+				<NavItem className={styles.subNavItem}
+					key={index}
+					label={item.label}
+					link={item.link}
+					isActive={router.pathname === item.link}
+					onClick={onItemSelected}/>
+			);
+		} else {
+			return (
+				<EmbeddedNavItem
+					label={item.label}
+					key={index}
+					isActive={isItemActive(item, router.pathname)}
+					onClick={(e) => {
+						e.stopPropagation();
+						selectEmbeddedNavItem(item);
+					}}
+				/>
+			);
 		}
-		/* eslint-disable */
-  }, [router.pathname, root]);
-
-  const subNav = subItems.map((item, index) => {
-    if (isNavigationItem(item)) {
-      return (
-        <NavItem className={styles.subNavItem}
-          key={index}
-          label={item.label}
-          link={item.link}
-          isActive={router.pathname === item.link}
-          onClick={onItemSelected}/>
-      );
-    } else {
-      return (
-        <EmbeddedNavItem
-          label={item.label}
-          key={index}
-          isActive={isItemActive(item, router.pathname)}
-          onClick={(e) => {
-            e.stopPropagation();
-            selectEmbeddedNavItem(item);
-          }}
-        />
-      );
-    }
-
-  });
-
-
-  return (
-    <li ref={optionsRef} className={classNames(isActive ? styles.hasNavItemActive : '', className)}>
-      <button
-        className={classNames(styles.subNavItemButton, { [styles.embedded]: !isRoot })}
-        onClick={() => isRoot ? setIsExpanded(!isExpanded) : popEmbeddedItem()}
-        aria-expanded={isExpanded}>
-        <span className={styles.subNavItemLabel} aria-current={isActive}>{label}</span>
-        <Icon className={isExpanded ? styles.subNavItemIconExpanded : styles.subNavItemIcon} name="angle-down"/>
-      </button>
-      {isExpanded &&
+
+	});
+
+	const isRoot = root.label === currentItem.label;
+
+	return (
+		<li ref={optionsRef} className={classNames(isActive ? styles.hasNavItemActive : '', className)}>
+			<button
+				className={classNames(styles.subNavItemButton, { [styles.embedded]: !isRoot })}
+				onClick={() => isRoot ? setIsExpanded(!isExpanded) : popEmbeddedItem()}
+				aria-expanded={isExpanded}>
+				<span className={styles.subNavItemLabel} aria-current={isActive}>{currentItem.label}</span>
+				<Icon className={isExpanded ? styles.subNavItemIconExpanded : styles.subNavItemIcon} name="angle-down"/>
+			</button>
+			{isExpanded &&
         <ul className={styles.subNavItemList} role="menu">
-          {subNav}
+        	{subNav}
         </ul>
-      }
-    </li>
-  );
+			}
+		</li>
+	);
+}
+
+function createPathToCurrentPageSubmenu(category: NavigationItemWithChildren, matcher: (element: NavigationItem) => boolean): NavigationItemWithChildren[] | null {
```

Comment:
elle est pas testée cette fonction ? elle est un peu complexe avec la recursivité, la boucle for, et les différents if ... ?

Reply:
Comme elle est privée, elle est testée avec le composant lui-même. Après, les tests sont pas hyper explicites mais ils testent de manière assez exhaustive donc 🤷 . J'ai prévu de repasser sur les fichiers de cette PR dans la suivante de nettoyage, si elle est pas trop grosse, j'inclue de renommer les tests

Reply:
ok parfait !